diff --git a/wscript_build.py b/wscript_build.py
index a470808..006334b 100644
--- a/wscript_build.py
+++ b/wscript_build.py
@@ -258,7 +258,7 @@ def build(ctx):
     ])
 
     subprocess_c = ctx.pick_first_matching_dep([
-        ( "osdep/subprocess-posix.c",            "posix" ),
+        ( "osdep/subprocess-posix.c",            "posix && !tvos" ),
         ( "osdep/subprocess-win.c",              "win32-desktop" ),
         ( "osdep/subprocess-dummy.c" ),
     ])